web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -167,55 +173,220 @@ file systems will be created on the selected device.");
 };
 
 /**
- * Allows to select LVM
+ * Form for configuring the system volume group.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID.
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {StorageDevice[]} [props.devices=[]] - Available storage devices.
+ * @param {onSubmitFn} [props.onSubmit=noop] - On submit callback.
+ * @param {onValidateFn} [props.onValidate=noop] - On validate callback.
+ *
+ * @callback onSubmitFn
+ * @param {string[]} devices - Name of the selected devices.
+ *
+ * @callback onValidateFn
+ * @param {boolean} valid
+ */
+const LVMSettingsForm = ({
+  id,
+  settings,
+  devices = [],
+  onSubmit: onSubmitProp = noop,
```

Comment:
NP: I see that the rename is because there is an internal `onSubmit` function. I'd just use `onSubmission` or **`handleSubmit`** for the internal one just for avoiding the rename. A personal taste, of course.